Packages/com.unity.inputsystem/InputSystem/Actions/InputActionSetupExtensions.cs

```diff
@@ -1564,19 +1564,26 @@ private BindingSyntax IteratePartBinding(bool next, string partName)
             /// that got erased was the last one in the array).
             /// </remarks>
             /// <exception cref="InvalidOperationException">The instance is not <see cref="valid"/>.</exception>
-            public void Erase()
+            public void Erase(bool onlyRemoveCompositeParts = false)
             {
                 if (!valid)
                     throw new InvalidOperationException("Instance not valid");
 
                 var isComposite = m_ActionMap.m_Bindings[m_BindingIndexInMap].isComposite;
-                ArrayHelpers.EraseAt(ref m_ActionMap.m_Bindings, m_BindingIndexInMap);
+                if (onlyRemoveCompositeParts && !isComposite)
+                    throw new InvalidOperationException("Binding should be a composite binding");
+                
+                var index = m_BindingIndexInMap;
+                if (!onlyRemoveCompositeParts)
+                    ArrayHelpers.EraseAt(ref m_ActionMap.m_Bindings, index);
+                else
+                    index++;
```

Comment:
Optional: I would maybe a comment here with something like in "Skips composite bindings parent index" (line 1580)